### PR TITLE
feat: allow disabling consensus WAL

### PR DESCRIPTION
**Pull request by deelawn**
This allows for the consensus state's WAL to be disabled.  This is desirable for testing against a running node when we don't care about the WAL -- it avoids potential permissions issues by not writing anything to disk.  This allows for our integration testing package to be imported and used outside of the gno repository.
<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯

Thank you for enabling this option 🙏

**Review by gfanton — COMMENTED** [review]
Besides that, I'm not sure why this file was initialized in `GNOROOT` by default instead of `GNOHOME`. Generally speaking, we should avoid initializing anything inside `GNOROOT` and `GNOROOT` should be treated as a read-only directory.

File: tm2/pkg/bft/consensus/config/config.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ type ConsensusConfig struct {
 	WalPath string `toml:"wal_file"`
 	walFile string // overrides WalPath if set
 
+	WALDisabled bool `toml:"wal_disabled" comment:"Disable the write-ahead log"`
```
Comment (thehowl) [inline tm2/pkg/bft/consensus/config/config.go:23]: I'd make this consistent with `WalPath` (suggest making that one `WALPath` following the convention); maybe place this next to `WalPath` in the struct?

Also I'm not sure whether this should be exposed in the config.

Comment (deelawn) [reply tm2/pkg/bft/consensus/config/config.go]: I've done what you suggested and removed the toml tag

File: tm2/pkg/bft/consensus/config/config.go [hunk 2]
```diff
@@ -16,9 +16,10 @@ const (
 // ConsensusConfig defines the configuration for the Tendermint consensus service,
 // including timeouts and details about the WAL and the block structure.
 type ConsensusConfig struct {
-	RootDir string `toml:"home"`
-	WalPath string `toml:"wal_file"`
-	walFile string // overrides WalPath if set
+	RootDir     string `toml:"home"`
+	WALPath     string `toml:"wal_file"`
+	WALDisabled bool
```
Comment (thehowl) [inline tm2/pkg/bft/consensus/config/config.go:21]:
```suggestion
	WALDisabled bool   `toml:"-"`
```

**Pull request merged**